packages/block-editor/src/components/block-toolbar/style.scss

```diff
@@ -76,6 +76,10 @@
 	.block-editor-block-toolbar {
 		overflow: auto;
 		overflow-y: hidden;
+		@include custom-scrollbars-on-hover($gray-600, $gray-600);
+
+		// Prevents padding from being applied to the left and right sides of the element.
+		scrollbar-gutter: auto;
```

Comment:
The custom scrollbar mixin has a style called [scrollbar-gutter: stable both-edges;](https://github.com/WordPress/gutenberg/blob/83e1a2c064c2233e8cd82dcccf4cd555c3497be1/packages/base-styles/_mixins.scss#L563). To prevent the button from being cut off due to this, we need to override it.

![image](https://github.com/WordPress/gutenberg/assets/54422211/b5832f63-bba7-49eb-99e4-5672962d11a1)